admin_solutions/authentication.adoc

```diff
@@ -5,10 +5,308 @@ toc::[]
+*_/etc/origin/master/master-config.yaml_* file into a new directory; for example:
++
+----
+$ mkdir basicauthconfig; cp master-config.yaml basicauthconfig
+----
+.. Not yet installed {product-title}, then start the {product-title} API server,
+specifying the hostname of the (future) {product-title} master and a directory
+to store the configuration file created by the start command:
++
+----
+$ openshift start master --public-master=<apiserver> --write-config=<directory>
+----
++
+For example:
++
+----
+$ openshift start master --public-master=https://myapiserver.com:8443 --write-config=basicauthconfig
+----
++
+[NOTE]
+====
+If you are installing with Ansible, then you must add the
+`*identityProvider*` configuration to the Ansible playbook.
+If you use the following steps to modify your configuration manually after installing with Ansible, then you will lose any modifications whenever you re-run the install tool or upgrade.
+====
++
+. Edit the new *_master-config.yaml_* file's `*identityProviders*` stanza.
+. Copy
+xref:../install_config/configuring_authentication.adoc#basic-auth-example-config[the example `*BasicAuthPasswordIdentityProvider*` configuration] and paste it to replace the existing stanza.
+. Make the following modifications to the `*identityProviders*` stanza:
+.. Change the provider `*name*` ("my_remote_basic_auth_provider") to something
+unique and relevant to your deployment. This name is prefixed to the returned
+user ID to form an identity name.
+.. If required,
+xref:../install_config/configuring_authentication.adoc#mapping-identities-to-users[change `*mappingMethod*`] to control how mappings are established between the
+provider's identities and user objects.
+.. Specify the `*url*` to use to connect to a server that accepts credentials in Basic authentication headers.
+.. Optionally, change the `*ca*` to the certificate bundle to use in order to validate server certificates for the configured URL.
+.. Optionally, change the `*certFile*` to the client certificate to present when making requests to the configured URL.
+.. If `*certFile*` is specified, then you must change the `*keyFile*` to the key for the client certificate.
+. Save your changes and close the file.
+. Start the {product-title} API server, specifying the configuration file you just
+modified:
++
+----
+$ openshift start master --config=<path/to/modified/config>/master-config.yaml
+----
+
+Once configured, any user logging in to the {product-title} web console will be
+prompted to log in using their Basic authentication credentials.
+
+[[basic-troubleshooting]]
+=== Troubleshooting
+
+[WARNING]
+====
+*[NEEDINFO]*: Is this troubleshooting information? It was in introduction of the install guide authentication topic, but seems out of place there. What purpose does this serve? Does any of it belong in the overview here, in other words does any of this need to be considered before configuring authentication?
```

Comment:
Most troubleshooting will probably be around network connectivity to the back end server.

Could include equivalent `curl` commands that could be run on the master for debugging. Something like this would be the equivalent of what the OpenShift API server would do, given the sample config. Replace "sampleuser:samplepassword" with a valid username/password to test a valid login, and replace with an invalid username/password to test an invalid login.

```
curl --cacert /path/to/ca.crt --cert /path/to/client.crt --key /path/to/client.key -u sampleuser:samplepassword -v https://www.example.com/remote-idp
```